darwin/Platform.c

```diff
@@ -124,6 +128,14 @@ void Platform_init(void) {
 #else
    Platform_timebaseToNS = 1.0;
 #endif
+
+   // Determine the number of clock ticks per second
+   errno = 0;
+   Platform_clockTicksPerSec = sysconf(_SC_CLK_TCK);
+
+   if (errno || Platform_clockTicksPerSec < 1) {
```

Comment:
Don't set/test errno - just test for the return value being (exactly) -1 to detect an error here (as documented).

Reply:
My documentation on `sysconf` reads a bit different in this section:

> RETURN VALUE
> The return value of sysconf() is one of the following:
>
> *  On error, -1 is returned **and errno is set to indicate the cause of the error** (for example, EINVAL, indicating that name is invalid).
>
> *  If name corresponds to a maximum or minimum limit, and that limit is indeterminate, -1 is returned and errno is not changed. **(To distinguish an indeterminate limit from an error, set errno to zero before the call, and then check whether errno is nonzero when -1 is returned.)**
>
> *  If name corresponds to an option, a positive value is returned if the option is supported, and -1 is returned if the option is not supported.
>
> *  Otherwise, the current value of the option or limit is returned. This value will not be more restrictive than the corresponding value that was described to the application in <unistd.h> or <limits.h> when the application was compiled.

(Emphasis mine)

What would be of discussion would be handling the case that this setting was indeterminate is correct, but given you'd have no value their either, the current check should yield sensible behaviour in all cases.

Reply:
| My documentation on sysconf [...]

Aha, I did not know that - thanks for the pointer.  Further complicating things, on Mac OS X this is the documentation:

https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man3/sysconf.3.html

I think its fine either way - personally, I'd stick with the simpler just-explicitly-test-minus-one like we do pretty much everywhere else, but I'm OK with the code as is here too.

Reply:
I'll do whatever you guys settle on. Trying to implement this little change (which involved things I understood) is bringing me to the edge of my knowledge/expertise :P

Reply:
As both on MacOS and Linux they agree on set `errno` on error and the check as given above is quite broad, I'd suggest to stick with this version. It both catches out-of-range values AND when the function reports any error; and does so consistently even when the sysconf implementation is inconsistent in this regard.

Reply:
@BenBE +1